pkg/server/server.go

```diff
@@ -87,6 +79,10 @@ type preShutdownHookEntry struct {
 
 // Run starts the KCP api-server. This function blocks until the api-server stops or an error.
 func (s *Server) Run(ctx context.Context) error {
+	if err := s.cfg.ClusterControllerOptions.Validate(); err != nil {
```

Comment:
Do you want to do this here, or only if we're installing the controller (L302)?

Reply:
I was preferring quick feedback for broken flags, but happy to move it down